arroyo-state/src/parquet.rs

```diff
@@ -265,7 +265,9 @@ impl BackingStore for ParquetBackend {
                     &operator_id,
                 ));
                 debug!("deleting {}", path);
-                storage_client.lock().await.delete(path).await?;
+                if let Err(e) = storage_client.lock().await.delete(path).await {
```

Comment:
Do we wanna check the error to make sure it is the "not present" error?

Reply:
Like this?
```
match storage_client.lock().await.delete(path).await {
    Err(StorageError::ObjectStore(object_store::Error::NotFound {
        path, ..
    })) => {
        warn!("Failed to delete file: {}", path);
    }
    Err(e) => return Err(anyhow!("Failed to delete file: {}", e)),
    _ => {}
}
```

We might want to put that into `StorageProvider::delete` instead.

Reply:
Yeah, given that S3 will never fail on a missing file, maybe it makes sense to adjust the object store API. The object_store docs say 
> If the object did not exist, the result may be an error or a success, depending on the behavior of the underlying store. For example, local filesystems, GCP, and Azure return an error, while S3 and in-memory will return Ok. If it is an error, it will be [`Error::NotFound`].

Reply:
Makes sense. I but that logic in the storage provider and renamed the function to `delete_if_present`